tree/tree/src/TTreeCache.cxx

```diff
@@ -614,10 +630,288 @@ Int_t TTreeCache::DropBranch(const char *bname, Bool_t subbranches /*= kFALSE*/)
+   Int_t *lbaskets   = b.GetBasketBytes();
+   Long64_t *entries = b.GetBasketEntry();
+   if (R__unlikely(!lbaskets || !entries)) {
+      //printf("No baskets or entries.\n");
+      return IOPos{0, 0};
+   }
+   //Int_t blistsize = b.GetListOfBaskets()->GetSize();
+   Int_t blistsize = b.GetWriteBasket();
+   if (R__unlikely(blistsize <= 0)) {
+      //printf("Basket list is size 0.\n");
+      return IOPos{0, 0};
+   }
+
+   // Search for the basket that contains the event of interest.  Unlike the primary cache, we
+   // are only interested in a single basket per branch - we don't try to fill the cache.
+   Long64_t basketOffset = TMath::BinarySearch(blistsize, entries, fTree->GetReadEntry());
+   if (basketOffset < 0) { // No entry found.
+      //printf("No entry offset found for entry %ld\n", fTree->GetReadEntry());
+      return IOPos{0, 0};
+   }
+
+   // Check to see if there's already a copy of this basket in memory.  If so, don't fetch it
+   if ((basketOffset < blistsize) &&
+           b.GetListOfBaskets()->UncheckedAt(basketOffset)) {
+
+       //printf("Basket is already in memory.\n");
+       return IOPos{0, 0};
+   }
+
+   Long64_t pos = b.GetBasketSeek(basketOffset);
+   Int_t len = lbaskets[basketOffset];
+   if (R__unlikely(pos <= 0 || len <= 0)) {
+      /*printf("Basket returned was invalid (basketOffset=%ld, pos=%ld, len=%d).\n", basketOffset, pos, len);
+      for (int idx=0; idx<blistsize; idx++) {
+         printf("Basket entry %d, first event %d, pos %ld\n", idx, entries[idx], b.GetBasketSeek(idx));
+      }*/
+      return IOPos{0, 0};
+    } // Sanity check
+   // Do not cache a basket if it is bigger than the cache size!
+   if (R__unlikely(len > fBufferSizeMin)) {
+      //printf("Basket size is greater than the cache size.\n");
+      return IOPos{0, 0};
+   }
+
+   return IOPos{static_cast<ULong64_t>(pos), static_cast<UInt_t>(len)};
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Given a particular IO description (offset / length) representing a 'miss' of
+/// the TTreeCache's primary cache, calculate all the corresponding IO that
+/// should be performed.
+///
+/// `all` indicates that this function should search the set of _all_ branches
+/// in this TTree.  When set to false, we only search through branches that
+/// have previously incurred a miss.
+///
+/// Returns:
+/// - TBranch pointer corresponding to the basket that will be retrieved by
+///   this IO operation.
+/// - If no corresponding branch could be found (or an error occurs), this
+///   returns nullptr.
+TBranch *TTreeCache::CalculateMissEntries(Long64_t pos_in, Int_t len_in, Bool_t all) {
+   if (R__unlikely((pos_in < 0) || (len_in < 0))) {
+      return nullptr;
+   }
+   auto pos = static_cast<ULong64_t>(pos_in);
+   auto len = static_cast<UInt_t>(len_in);
+
+   int count = all ? (fTree->GetListOfLeaves())->GetEntriesFast() : fMissCache->fBranches.size();
```

Comment:
Not clear to me there's an easy way to do this:

  - If `all == false`, then we are iterating across a `std::vector<Branch*>`.
  - If `all == true`, then we are iterating over a `TObjArray` of `TLeaf*`.

Hence, one needs a little bit of logic per loop iteration to get to the correct object.